ui/desktop/src/main.ts

```diff
@@ -903,6 +887,60 @@ app.whenReady().then(async () => {
   });
 });
 
+/**
+ * Fetches the allowed extensions list from the remote YAML file if GOOSE_ALLOWLIST is set.
+ * If the ALLOWLIST is not set, any are allowed. If one is set, it will warn if the deeplink 
+ * doesn't match a command from the list. 
+ * If it fails to load, then it will return an empty list.
+ * If the format is incorrect, it will return an empty list.
+ * Format of yaml is:
+ *  
+ ```yaml:
+ extensions:
```

Comment:
idea to consider in future: make the format of this the same as the yaml blocks we use in `~/.config/goose/config.yaml` so we can compare extension configs as a whole.

Reply:
yeah could do that.